[INIT/#1] initial commit

angryPodo opened the pull request:
## Related issue 🛠
- closed #1 

## Work Description ✏️
- 레포 세팅 다시하기
- 패키지 경로도 다 수정
- 오타 개많은거 수정

졸면서 코딩하지말자

Hyobeen-Park commented on .github/workflows/pr_checker.yml:
```diff
@@ -0,0 +1,51 @@
+          DEV_BASE_URL: ${{ secrets.DEV_BASE_URL }}
+          PROD_BASE_URL: ${{ secrets.PROD_BASE_URL }}
+        run: |
+          echo sdk.dir=$ANDROID_SDK_ROOT >> local.properties
```
P3: 요거 필요한가요??

Hyobeen-Park commented on README.md:
```diff
@@ -0,0 +1 @@
+# 머요
```
### 머요

Hyobeen-Park commented on app/build.gradle.kts:
```diff
@@ -0,0 +1,76 @@
+
+    // other dependencies
+    implementation(libs.timber)
+    implementation(libs.androidx.appcompat)
```
P3: appcompat이 app에 필요한가요??? (just wonder)

Hyobeen-Park commented on app/src/main/java/com/hilingual/App.kt:
```diff
@@ -0,0 +1,18 @@
+
+@HiltAndroidApp
+class App : Application() {
+    override fun onCreate() {
+        super.onCreate()
+
+        initTimber()
+    }
```
P3: 다크모드 막을래 말래

Hyobeen-Park commented on core/designsystem/build.gradle.kts:
```diff
@@ -0,0 +1,23 @@
+    // core
+    implementation(projects.core.common)
+
+    // androidx
+    implementation(libs.bundles.androidx)
```
P3: compose 플러그인 안에 들어있는 것 외에 또 필요한 androidx가 있나요?

Hyobeen-Park commented on app/build.gradle.kts:
```diff
@@ -0,0 +1,76 @@
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+
+        buildConfigField(
+            "String",
+            "BASE_URL",
+            properties.getProperty("dev.base.url")
+        )
```
P3: network 모듈에서 지정해주고 있어서 얘는 없어도 되지 않을까요?

Hyobeen-Park reviewed (approved):
LGTM🚀🚀 빌드로직 어려웠을텐데 세팅하느라 수고했어요!!

nhyeonii reviewed (approved):
고생하셨습니다 !!! 구조 꼼꼼히 읽고 리뷰 남기겠습니다

Daljyeong commented on settings.gradle.kts:
```diff
@@ -0,0 +1,47 @@
+}
+
+rootProject.name = "Hi-lingual"
+// app
+include(":app")
+
+// core
+include(":core:navigation")
+include(":core:common")
+include(":core:designsystem")
+include(":core:network")
+include(":core:localstorage")
+
+// data
+include(":data:auth")
+include(":data:user")
+include(":data:diary")
+include(":data:voca")
+include(":data:calendar")
+
+// presentation
+include(":presentation:home")
+include(":presentation:auth")
+include(":presentation:diarywrtie")
+include(":presentation:diaryfeedback")
+include(":presentation:voca")
+include(":presentation:main")
```
P3: 멀티 모듈을 공부하다보니 아래 코드를 settings.gradle.kts 파일을 추가하면 로컬 모듈을 type-safe하게 참조할 수 있다는데, 따로 사용하지 않은 이유가 있는 걸까요? 모듈 수가 많지 않아서 이번 플젝에선 굳이 사용하지 않아도 되는 건지 궁금합니닷
```
enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")
```

[레퍼런스 링크]
[https://dev-inventory.com/56?category=1192367#4.%20%EB%AA%A8%EB%93%88%20%EA%B4%80%EA%B3%84%20%EC%A0%95%EC%9D%98-1-3](url)

Daljyeong reviewed (commented):
수고하셨습니당!! 역시 리드 체고 🔥

angryPodo committed:
[INIT/#1] initial commit

angryPodo committed:
[INIT/#1] initial commit

nahy-512 commented on core/designsystem/src/main/java/com/hilingual/core/designsystem/theme/Color.kt:
```diff
@@ -0,0 +1,11 @@
+package com.hilingual.ui.theme
```
P2: theme 이쪽 파일들 
```suggestion
package com.hilingual.core.designsystem.theme
```
이렇게 바꾸라고 뜨는듯!

angryPodo replied:
아 맞네요 수정하겟굽

nahy-512 commented on core/network/build.gradle.kts:
```diff
@@ -0,0 +1,47 @@
+
+dependencies {
+    //core
+    implementation(projects.core.localstorage)
```
P3: 네트워크 모듈에 이 의존성이 있는 건 인터셉터에서 토큰을 넣어주기 위함일까요?

angryPodo replied:
네 그리고 리프레쉬나 엑세스를 받아서 저장하기 위함도 잇습니다

angryPodo committed:
[INIT/#1] initial commit

nahy-512 commented on gradle/libs.versions.toml:
```diff
@@ -0,0 +1,214 @@
+kotlin-compose = { id = "org.jetbrains.kotlin.plugin.compose", version.ref = "kotlin" }
+kotlin-serialization = { id = "org.jetbrains.kotlin.plugin.serialization", version.ref = "kotlin" }
+jetbrains-kotlin-jvm = { id = "org.jetbrains.kotlin.jvm", version.ref = "jetbrainsKotlinJvm" }
+compose-compiler = { id = "org.jetbrains.kotlin.plugin.compose", version.ref = "composeCompilerGradlePlugin" }
```
P3: 이거 154, 157번째 줄 `kotlin-compose`랑 `compose-compiler` 겹쳐서
```There are multiple dependencies org.jetbrains.kotlin.plugin.compose but with different version```
오류 뜨는데 둘이 뭐가 다른 거예요??

angryPodo replied:
아 같은 내용이 두번 들어가긴했네요 근데 오류나서 빌드 안대나요? 왜 나는 빌드 잘대냐

nahy-512 replied:
빌드는 잘 되어요 근데 밑줄 뜨길래 달아봤어요

angryPodo replied:
아하 글쿤용 사실 같은 내용이고 버전 참조가 달라서 오류가 났을거에요 버저닝을 같이 가져가는 방식이랑 독립적으로 관리하는게 잇어요

nahy-512 reviewed (approved):
고생하셨습니다!

nahy-512 commented on gradle/libs.versions.toml:
```diff
@@ -0,0 +1,214 @@
+jakewharton-process-phoenix = { group = "com.jakewharton", name = "process-phoenix", version.ref = "phoenix" }
+pebble = { group = "io.github.chattymin", name = "pebble", version.ref = "pebble" }
+
+# Room
+room-runtime = { group = "androidx.room", name = "room-runtime", version.ref = "room" }
+room-ktx = { group = "androidx.room", name = "room-ktx", version.ref = "room" }
+room-compiler = { group = "androidx.room", name = "room-compiler", version.ref = "room" }
```
P3: 단어장 룸디비 안 쓰기로 결정됐지만 일단 넣어두기??

angryPodo replied:
왜냐면 로컬 스토리지에 대한건 언제든 쓸수잇으니깐!

Hyobeen-Park replied:
흠흠 일단 사용하지 않는거라면 빼는거 어떠신지요?!?? 추후에 room이 필요할 때 추가하는게 더 좋지 않을까 하는 의견 한번 내봅니당ㅎㅎ

angryPodo replied:
왜 나 캘린더 캐싱할거야

Hyobeen-Park replied:
오호라 인정

nhyeonii commented on presentation/voca/src/test/java/com/hilingual/presentation/voca/ExampleUnitTest.kt:
```diff
@@ -0,0 +1,17 @@
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleUnitTest {
```
P3 : 저희 어제 모각작때 자동 생성 기본 템플릿 테스트 코드는 삭제하기로 하지 않앗나요..?! 궁금해서 물어봅니당 👀

angryPodo replied:
언더스코어 오류 때문에 삭제하려고 했던건데 해결해서 냅둡니당. 테코 작성을 할수도 있으니!

nhyeonii reviewed (approved):
수고하셨습니다 !!

angryPodo committed:
[INIT/#1] Relocate theme files to designsystem module

The theme-related files (Color.kt, Type.kt, Theme.kt) have been moved from the `presentation/main` module to the `core/designsystem` module.

This change centralizes the design system components, improving modularity and maintainability.

The import statements in `MainActivity.kt` have been updated to reflect the new location of `HilingualTheme`. Additionally, the unused `kotlin-compose` plugin has been removed from `libs.versions.toml`.
